src/architecture/_GeneralModuleFiles/py_sys_model.i

```diff
@@ -13,49 +30,158 @@ import sys
+            for name in dir(base):
+                if name.startswith("_"):
+                    continue
+                if name in ("__init__",):
+                    continue
+                candidateNames.add(name)
+
+        for name in candidateNames:
+            subAttr = getattr(cls, name, None)
+            if not callable(subAttr):
+                continue
+
+            # Find the first base that defines this name
+            baseAttr = None
+            for base in cppBases:
+                if hasattr(base, name):
+                    baseAttr = getattr(base, name)
+                    break
+
+            if baseAttr is None:
+                continue
+
+            # Only wrap if the subclass actually overrides the method
+            if subAttr is baseAttr:
+                continue
+
+            # Avoid double wrapping
+            if getattr(subAttr, "_sysmodelLogged", False):
+                continue
+
+            setattr(cls, name, logError(subAttr))
 
     def __init_subclass__(cls):
-        # Make it so any exceptions in UpdateState and Reset
-        # print any exceptions before returning control to
-        # C++ (at which point exceptions will crash the program)
-        cls.UpdateState = logError(cls.UpdateState)
-        cls.Reset = logError(cls.Reset)
+        """Hook that runs for every Python subclass of SysModelMixin.
+
+        It:
+          * wraps all overridden director methods from SWIG C++ proxy bases
+            in the MRO with logError
+          * enforces super().__init__() on subclasses
+        """
+        super().__init_subclass__()
+
+        # Wrap overridden director methods
+        cls._wrapOverriddenDirectorMethods()
+
+        origInit = cls.__init__
+
+        # Avoid double wrapping if someone subclasses a subclass.
+        if getattr(origInit, "_sysmodelWrappedInit", False):
+            return
+
+        def wrappedInit(self, *args, **kwargs):
+            # Clear flag before running user __init__.
+            object.__setattr__(self, "_sysmodelBaseInitSeen", False)
+            origInit(self, *args, **kwargs)
+            if not getattr(self, "_sysmodelBaseInitSeen", False):
+                errorMsg = (
+                    "Need to call parent __init__ in SysModel based subclasses:\n"
+                    f"class {cls.__name__}(...):\n"
+                    "    def __init__(...):\n"
+                    "        super().__init__()"
+                )
+                raise SyntaxError(errorMsg)
+
+        wrappedInit._sysmodelWrappedInit = True
+        cls.__init__ = wrappedInit
+
+
+class SysModel(SysModelMixin, _SysModel):
+    """Python friendly base class for Basilisk system models.
+
+    Inherits:
+      * SysModelMixin  (pure Python)
+      * _SysModel      (SWIG proxy for C++ SysModel)
+
+    Typical usage for new models:
+        from Basilisk.architecture import sysModel
+
+        class MyModel(sysModel.SysModel):
+            def __init__(self):
```

Comment:
The `__init__` method in this documentation looks different than what we have in https://avslab.github.io/basilisk/Learn/makingModules/pyModules.html.  Should we update `pyModules.rst` documentation?